serializer_impl.hh

```diff
@@ -800,6 +800,42 @@ public:
 private:
     input_stream _in;
     size_t _size;
+    utils::chunked_vector<input_stream> _substreams;
+
+    void fill_substreams() requires (!IsForward) {
+        input_stream in = _in;
+        input_stream in2 = _in;
+        for (size_t i = 0; i < size(); ++i) {
+            size_t old_size = in.size();
+            serializer<T>::skip(in);
+            size_t new_size = in.size();
+
+            _substreams.push_back(in2.read_substream(old_size - new_size));
+        }
+    }
+
+    struct forward_iterator_data {
+        input_stream _in = simple_input_stream();
+        void skip() {
+            serializer<T>::skip(_in);
+            // auto len = read_frame_size();
+            // _in.skip(len);
+        }
+        value_type deserialize_next() {
+            return deserialize(_in, boost::type<T>());
+        }
+    };
+    struct reverse_iterator_data {
+        std::reverse_iterator<utils::chunked_vector<input_stream>::const_iterator> _substream_it;
+        void skip() {
+            _substream_it++;
+        }
+        value_type deserialize_next() {
+            input_stream is = *(_substream_it++);
```

Comment:
nit: it would be a little more efficient to do the following to eliminate copy of _substream_it:
```c++
    input_stream is = *_substream_it;
    ++_substream_it;
```

Reply:
I do it on purpose - otherwise it is impossible to use the same vector deserializer twice, as the substreams which are held are exhausted. If this is not a concern, I can make this modification.

Reply:
I'm not sure I understand.
We copy the substream to an automatic variable `is`
and the iterator is incremented in any case.
Why would my suggestion exhaust the substreams?

Reply:
No, you're correct